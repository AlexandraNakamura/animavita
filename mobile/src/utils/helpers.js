import AsyncStorage from '@react-native-community/async-storage';

export const sanitizeUsername = (name) => {
  if (name && name.length > 8) {
    return `\n ${name}`;
  }

  return name;
};

export const isEmpty = array => array instanceof Array && array.length === 0;

export const getToken = async () => JSON.parse(await AsyncStorage.getItem('@animavita:authToken')) || {};
