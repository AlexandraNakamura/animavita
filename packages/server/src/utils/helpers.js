import AWS from 'aws-sdk';

require('dotenv').config();

const {
  S3_KEY, S3_SECRET, S3_BUCKET, S3_REGION
} = process.env;

export const getAnimalGender = (gender) => {
  const genders = [
    {
      key: 'male',
      translate: 'Macho'
    },
    {
      key: 'female',
      translate: 'Fêmea'
    }
  ];

  return genders.find(element => element.key === gender).translate;
};

export const getAnimalSize = (size) => {
  const sizes = [
    {
      key: 'small',
      translate: 'Pequeno'
    },
    {
      key: 'medium',
      translate: 'Médio'
    },
    {
      key: 'bigger',
      translate: 'Grande'
    }
  ];

  return sizes.find(element => element.key === size).translate;
};

export const uploadFile = async (upload) => {
  let url = '';
  // eslint-disable-next-line new-cap
  const base64Data = new Buffer.from(
    upload.preview.uri.replace(/^data:image\/\w+;base64,/, ''),
    'base64'
  );

  const params = {
    Bucket: S3_BUCKET,
    Key: upload.image.name,
    Body: base64Data,
    ACL: 'public-read',
    ContentEncoding: 'base64',
    ContentType: upload.image.type
  };

  const s3 = new AWS.S3({
    accessKeyId: S3_KEY,
    secretAccessKey: S3_SECRET,
    region: S3_REGION
  });

  try {
    const { Location } = await s3.upload(params).promise();
    url = Location;
  } catch (err) {
    throw err;
  }

  return url;
};
