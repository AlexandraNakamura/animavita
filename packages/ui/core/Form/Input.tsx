import React from 'react';
import {TextInputProps} from 'react-native';
import {px2ddp, PossibleThemes, useTheme} from '@animavita/theme';
import styled from 'styled-components/native';

// styles
export const StyledTextInput = styled.TextInput<{themeName: PossibleThemes}>`
  flex: 1;
  padding: ${px2ddp(2)}px 0px;
  color: ${({themeName, theme}) => (themeName === 'light' ? theme.black : theme.white)};
`;

// types
type InputProps = TextInputProps;

const Input: React.FC<InputProps> = ({...props}) => {
  const {themeName} = useTheme();
  return <StyledTextInput themeName={themeName} {...props} />;
};

export default Input;
