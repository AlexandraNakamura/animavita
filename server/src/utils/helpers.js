export const getAnimalGender = (gender) => {
  const genders = [
    {
      key: 'male',
      trans: 'Macho'
    },
    {
      key: 'female',
      trans: 'Fêmea'
    }
  ];

  return genders.find(element => element.key === gender).trans;
};

export const getAnimalSize = (size) => {
  const sizes = [
    {
      key: 'small',
      trans: 'Pequeno'
    },
    {
      key: 'medium',
      trans: 'Médio'
    },
    {
      key: 'bigger',
      trans: 'Grande'
    }
  ];

  return sizes.find(element => element.key === size).trans;
};
