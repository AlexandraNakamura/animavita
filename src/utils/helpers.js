import AsyncStorage from '@react-native-community/async-storage';

export const sanitizeUsername = (name) => {
  if (name && name.length > 8) {
    return `\n ${name}`;
  }

  return name;
};

export const isEmpty = (array) => {
  if (array instanceof Array && array.length === 0) {
    return true;
  }

  return false;
};

export const getUser = async () => JSON.parse(await AsyncStorage.getItem('@animativa:user'));
