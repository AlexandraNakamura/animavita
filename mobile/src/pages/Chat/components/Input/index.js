import React, { Component } from "react";

import {
  View,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  Text
} from "react-native";
import { THEME_COLORS } from "~/utils/constants";
import { Send } from 'react-native-gifted-chat';

class Input extends Component {
  state = {
    message: ''
  }

  render() {
    return (
      <View style={styles.inputContainer}>
        <TextInput
          value={this.state.message}
          onChangeText={message => this.setState({ message })}
          style={styles.input}
          underlineColorAndroid="rgba(0,0,0,0)"
        />
          <Send {...this.props}>

            <Text style={styles.button}>Enviar</Text>
          </Send>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  inputContainer: {
    height: 42,
    paddingHorizontal: 10,
    backgroundColor: "#fafafa",
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: "#ccc",
    flexDirection: "row",
    alignItems: "center"
  },

  input: {
    flex: 1,
    height: 30,
    paddingHorizontal: 10,
    paddingVertical: 0,
    backgroundColor: "#FFF",
    borderWidth: 1,
    borderColor: "#ddd",
    borderRadius: 12
  },

  button: {
    marginLeft: 10,
    color: THEME_COLORS.SECONDARY,
    fontWeight: "bold"
  }
});



export default Input