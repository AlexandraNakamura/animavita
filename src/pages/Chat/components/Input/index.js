import React, { Component } from "react";

import {
  View,
  TextInput,
  StyleSheet,
  TouchableOpacity,
  Text
} from "react-native";
import { THEME_COLORS } from "~/utils/constants";

class Input extends Component {
  state = {
    message: ""
  };

  handleAddMessage = async () => {
    const { author } = this.props;
    const { message } = this.state;

  
    this.setState({ message: "" });
  };

  render() {
    return (
      <View style={styles.inputContainer}>
        <TextInput
          value={this.state.message}
          onChangeText={message => this.setState({ message })}
          style={styles.input}
          underlineColorAndroid="rgba(0,0,0,0)"
        />
        <TouchableOpacity onPress={this.handleAddMessage}>
          <Text style={styles.button}>Enviar</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  inputContainer: {
    height: 42,
    paddingHorizontal: 10,
    backgroundColor: "#fafafa",
    borderTopWidth: StyleSheet.hairlineWidth,
    borderTopColor: "#ccc",
    flexDirection: "row",
    alignItems: "center"
  },

  input: {
    flex: 1,
    height: 30,
    paddingHorizontal: 10,
    paddingVertical: 0,
    backgroundColor: "#FFF",
    borderWidth: 1,
    borderColor: "#ddd",
    borderRadius: 12
  },

  button: {
    marginLeft: 10,
    color: THEME_COLORS.SECONDARY,
    fontWeight: "bold"
  }
});



export default Input