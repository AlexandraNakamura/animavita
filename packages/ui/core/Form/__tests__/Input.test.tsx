import React from 'react';
import {fireEvent} from '@testing-library/react-native';
import renderer from 'react-test-renderer';

import {Mount, ThemeContext} from '../../../tests/helpers';
import Input from '../Input';

describe('Input', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('calls onChangeText', () => {
    const cb = jest.fn();

    const {getByLabelText} = Mount(<Input onChangeText={cb} accessibilityLabel="username" />);

    fireEvent.changeText(getByLabelText('username'), 'a');

    expect(cb).toHaveBeenCalledWith('a');
  });

  it('renders correctly', () => {
    const cb = jest.fn();
    const tree = renderer.create(ThemeContext(<Input onChangeText={cb} accessibilityLabel="username" />)).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
