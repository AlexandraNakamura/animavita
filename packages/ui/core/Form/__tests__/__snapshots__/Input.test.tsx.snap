// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input renders correctly 1`] = `
<TextInput
  accessibilityLabel="username"
  allowFontScaling={true}
  onChangeText={[MockFunction]}
  rejectResponderTermination={true}
  style={
    Array [
      Object {
        "color": "#353434",
        "paddingBottom": 4,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 4,
        "width": "100%",
      },
    ]
  }
  themeName="light"
  underlineColorAndroid="transparent"
/>
`;
